### ISSUE TITLE
How to set authentication for MQTT broker

### ISSUE THREAD

**Issue by anselanza** [pabsan-0/libnvds_mqtt_proto#1]
Thank you for this extremely useful library!

I am able to get everything working as expected for my use case (I am doing inference using a custom model on a camera stream), BUT this only works when connecting to a MQTT Broker that requires no username/password. How do I set the username/password details if I want to connect to a broker that *does* require these?

In my DeepStream config, I have the following under the `sink1` section:
```
[sink1]
enable=1
type=6
msg-broker-proto-lib=/opt/nvidia/deepstream/deepstream/lib/libnvds_mqtt_proto.so
msg-broker-conn-str=localhost;1883
msg-conv-config=config_messages.txt
topic=detections
msg-conv-payload-type=0
disable-msgconv=0
#msg-broker-config=config_broker.txt
```

The `msg-broker-conn-str` seems to be able to specify `host;port;name` only - no username and/or password. And yet I cannot remove this line.

I have commented out `msg-broker-config` for now because it seems to be ignored anyway. I tried including the following in `config_broker.txt`:

```
[message-broker]
conn-str=192.168.27.100;1883
hostname=192.168.27.100
username=someusername
password=somepassword
port=1883
topic=detections
```
Nothing in this file seems to be used anyway. Even intentional mistakes don't produce any errors. Yet I see no other way to configure authentication details. What am I missing?

**Comment by pabsan-0** [comment]
Hi!

It's been a looong time I don't look at this repo, I don't even think that password login is implemented.

To achieve it, you'd need to dive into the code in this repo and write it yourself. All MQTT is handled by an external library, see the includes, then check out its documentation to figure password-protected connections. Have a look in the script to see what is implemented at all ;) 

I suggest you hardcode the password in the script for tests then modify the code to receive it via parsing the connection string.

Also, IIRC newer Deepstream versions have native MQTT support. While I love that my repo is getting attention, take a look at the official docs for it might save you some headaches.

Cheers

**Comment by anselanza** [comment]
Yes, newer DeepStream (from 6.3) does have built-in MQTT support, but the Jetson Nano only supports up to DeepStream 6.0.1.

This is why your library remains useful.
